ksql-cli/src/main/java/io/confluent/ksql/cli/console/KsqlLineParser.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.ksql.cli.console;
+
+import java.util.Objects;
+import java.util.function.Function;
+import org.jline.reader.EOFError;
+import org.jline.reader.ParsedLine;
+import org.jline.reader.Parser;
+
+/**
+ * Ensures complete lines are either terminated with a semi-colon or are cli commands.
+ */
+final class KsqlLineParser implements Parser {
+
+  private static final String TERMINATION_CHAR = ";";
+
+  private final Parser delegate;
+  private final Function<String, Boolean> cliCmdPredicate;
```

Comment:
nit: why not use Predicate<String> here?

Reply:
Good point! ;)